ci/build_wheel_python.sh

```diff
@@ -17,7 +17,14 @@ CPP_WHEELHOUSE=$(RAPIDS_PY_WHEEL_NAME="libkvikio_${RAPIDS_PY_CUDA_SUFFIX}" rapid
 
 cd "${package_dir}"
 
-python -m pip wheel . -w dist -vvv --no-deps --disable-pip-version-check --find-links ${CPP_WHEELHOUSE}
+# ensure 'kvikio' wheel builds always use the 'libkvikio' just built in the same CI run
+#
+# using env variable PIP_CONSTRAINT is necessary to ensure the constraints
+# are used when creating the isolated build environment
+echo "libkvikio-${RAPIDS_PY_CUDA_SUFFIX} @ file://$(echo ${CPP_WHEELHOUSE}/libkvikio_*.whl)" > ./constraints.txt
+
+PIP_CONSTRAINT="${PWD}/constraints.txt" \
+    python -m pip wheel . -w dist -vvv --no-deps --disable-pip-version-check
```

Comment:
Am curious whether just adding `--no-index` would be sufficient for us

```suggestion
python -m pip wheel . -w dist -vvv --no-deps --disable-pip-version-check --no-index --find-links ${CPP_WHEELHOUSE}
```

Reply:
Thanks for the suggestion! I hadn't considered this.

But I wouldn't expect that to be sufficient here. We're building with build isolation, and this `${CPP_WHEELHOUSE}` directory doesn't contain wheels for other build dependencies like `scikit-build-core` and `rapids-build-backend`.

So to use `--no-index --find-links ${CPP_WHEELHOUSE}` we'd have to also do something like `pip download` the other build dependencies into that local directory. I think the approach with constraints is simpler for as long as we want to use build isolation (which, per https://github.com/rapidsai/build-planning/issues/69#issuecomment-2187054141, might not be fore long).

Reply:
Discussed offline with James. We decided James will write up an issue for follow up and we will skip this suggestion in this PR

Reply:
Thanks for talking through it with me. Here's the issue: https://github.com/rapidsai/build-planning/issues/79